iocore/cache/test/test_Alternate_L_to_S_remove_L.cc

```diff
@@ -0,0 +1,259 @@
+/** @file
+
+  A brief file description
+
+  @section license License
+
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#define LARGE_FILE 10 * 1024 * 1024
+#define SMALL_FILE 10 * 1024
+
+#include "main.h"
+
+// delete dir
+Dir dir = {};
+
+class CacheAltReadAgain2 : public CacheTestHandler
+{
+public:
+  CacheAltReadAgain2(size_t size, const char *url) : CacheTestHandler()
+  {
+    auto rt = new CacheReadTest(size, this, url);
+
+    rt->mutex = this->mutex;
+
+    rt->info.destroy();
+
+    rt->info.create();
+    build_hdrs(rt->info, url, "application/x-javascript");
+
+    this->_rt = rt;
+
+    SET_HANDLER(&CacheAltReadAgain2::start_test);
+  }
+
+  int
+  start_test(int event, void *e)
+  {
+    REQUIRE(event == EVENT_IMMEDIATE);
+    this_ethread()->schedule_imm(this->_rt);
+    return 0;
+  }
+
+  virtual void
+  handle_cache_event(int event, CacheTestBase *base)
+  {
+    switch (event) {
+    case CACHE_EVENT_OPEN_READ:
+      base->do_io_read();
+      break;
```

Comment:
Did you miss "validate_content_type()" here ?

Reply:
will add it.